x/bank/simulation/msgs.go

```diff
@@ -34,26 +53,124 @@ func SingleInputSendTx(mapper auth.AccountKeeper) simulation.Operation {
+	}
+
+	action = fmt.Sprintf("%s is sending %s %s to %s",
+		fromAcc.Address.String(),
+		amt.String(),
+		initFromCoins[denomIndex].Denom,
+		toAddr.String(),
+	)
+
+	coins := sdk.Coins{sdk.NewCoin(initFromCoins[denomIndex].Denom, amt)}
+	msg = bank.NewMsgSend(fromAcc.Address, toAcc.Address, coins)
+	return
+}
+
+// Sends and verifies the transition of a msg send.
+func sendAndVerifyMsgSend(app *baseapp.BaseApp, mapper auth.AccountKeeper, msg bank.MsgSend, ctx sdk.Context, privkeys []crypto.PrivKey, handler sdk.Handler) error {
+	fromAcc := mapper.GetAccount(ctx, msg.FromAddress)
+	AccountNumbers := []uint64{fromAcc.GetAccountNumber()}
+	SequenceNumbers := []uint64{fromAcc.GetSequence()}
+	initialFromAddrCoins := fromAcc.GetCoins()
+
+	toAcc := mapper.GetAccount(ctx, msg.ToAddress)
+	initialToAddrCoins := toAcc.GetCoins()
+
+	if handler != nil {
+		res := handler(ctx, msg)
+		if !res.IsOK() {
+			if res.Code == bank.CodeSendDisabled {
+				return nil
+			}
+			// TODO: Do this in a more 'canonical' way
+			return fmt.Errorf("handling msg failed %v", res)
+		}
+	} else {
+		tx := mock.GenTx([]sdk.Msg{msg},
+			AccountNumbers,
+			SequenceNumbers,
+			privkeys...)
+		res := app.Deliver(tx)
+		if !res.IsOK() {
+			// TODO: Do this in a more 'canonical' way
+			return fmt.Errorf("Deliver failed %v", res)
+		}
+	}
+
+	fromAcc = mapper.GetAccount(ctx, msg.FromAddress)
+	toAcc = mapper.GetAccount(ctx, msg.ToAddress)
+
+	if !initialFromAddrCoins.Minus(msg.Amount).IsEqual(fromAcc.GetCoins()) {
+		return fmt.Errorf("fromAddress %s had an incorrect amount of coins", fromAcc.GetAddress())
+	}
+
+	if !initialToAddrCoins.Plus(msg.Amount).IsEqual(toAcc.GetCoins()) {
+		return fmt.Errorf("toAddress %s had an incorrect amount of coins", toAcc.GetAddress())
+	}
+
+	return nil
+}
+
+// SingleInputSendTx tests and runs a single msg multisend w/ auth, with one input and one output, where both
+// accounts already exist.
+func SingleInputMultiSendTx(mapper auth.AccountKeeper) simulation.Operation {
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account, event func(string)) (action string, fOps []simulation.FutureOperation, err error) {
+		fromAcc, action, msg, abort := createSingleInputMultiSendMsg(r, ctx, accs, mapper)
+		if abort {
+			return action, nil, nil
+		}
+		err = sendAndVerifyMsgMultiSend(app, mapper, msg, ctx, []crypto.PrivKey{fromAcc.PrivKey}, nil)
+		if err != nil {
+			return "", nil, err
+		}
+		event("bank/sendAndVerifyTxMultiSend/ok")
+
+		return action, nil, nil
+	}
+}
+
+// SingleInputSendMsg tests and runs a single msg multisend, with one input and one output, where both
 // accounts already exist.
-func SingleInputSendMsg(mapper auth.AccountKeeper, bk bank.Keeper) simulation.Operation {
+func SingleInputMultiSendMsg(mapper auth.AccountKeeper, bk bank.Keeper) simulation.Operation {
```

Comment:
for the sake of consistency:

```suggestion
func SingleInputMsgMultiSend(mapper auth.AccountKeeper, bk bank.Keeper) simulation.Operation {
```